e2e-tests/src/test/button_game/mod.rs

```diff
@@ -71,14 +71,18 @@ pub fn simple_dex(config: &Config) -> Result<()> {
         ],
     )?;
 
+    assert!(
+        dex.out_given_in(account_conn, token1, token2, 100).is_ok(),
+        "out_given_in should always return"
+    );
```

Comment:
This doesn't really make sense - the `Err` variant from this call signifies some connection or serialization error, not an `Err` variant received from the contract call. If the contract returns a `Result`, then a `contract_read` will return `Result<Result<_>>`.

```suggestion
```